src/popup/components/Results.tsx

```diff
@@ -0,0 +1,26 @@
+import React, { FC, useEffect, useState } from "react";
+import { TotpResponse, isCode } from "../../types";
+
+export interface ResultsProps {
+  searchKey: string;
+}
+
+export const Results: FC<ResultsProps> = (props: ResultsProps) => {
+  const [result, setResult] = useState("");
+
+  const handleResult = (response: TotpResponse): void => {
+    if (response == null) {
+      setResult("null");
+    } else if (isCode(response)) {
+      setResult(response.code);
+    } else {
+      setResult(response.error);
+    }
+  };
+
+  useEffect(() => {
+    chrome.runtime.sendMessage(props.searchKey, handleResult);
+  });
```

Comment:
It would probably make sense to add `}, [props.searchKey]);` as a dependency, so that if the prop does not change, the message does not get sent again.
It's in this app not a big deal though, because everything would work the same in that case too.